Qt.py

```diff
@@ -891,107 +607,378 @@ def _cli(args):
+    return types.ModuleType(__name__ + "." + name)
 
-"""
 
-QtCompat._cli = _cli
-QtCompat._convert = _convert
+def _setup(module, extras):
+    """Install common submodules"""
 
+    Qt.__binding__ = module
 
-"""Apply strict mode
+    for name in _common_members.keys() + extras:
+        try:
+            submodule = importlib.import_module(module + "." + name)
+        except ImportError:
+            continue
 
-This make Qt.py into a subset of PySide2 members that exist
-across all other bindings.
+        # Store reference to original binding
+        setattr(Qt, name, _new_module(name))
+        setattr(Qt, "_" + name, submodule)
 
-"""
 
-for module, members in _strict_members.items():
-    for member in members:
-        orig = getattr(sys.modules[__name__], "_%s" % module)
-        repl = getattr(sys.modules[__name__], module)
-        setattr(repl, member, getattr(orig, member))
+def _pyside2():
+    """Initialise PySide2
 
+    These functions serve to test the existence of a binding
+    along with set it up in such a way that it aligns with
+    the final step; adding members from the original binding
+    to Qt.py
 
-# Enable direct import of submodules
-# E.g. import Qt.QtCore
-sys.modules.update({
-    __name__ + ".QtGui": QtGui,
-    __name__ + ".QtCore": QtCore,
-    __name__ + ".QtWidgets": QtWidgets,
-    __name__ + ".QtXml": QtXml,
-    __name__ + ".QtNetwork": QtNetwork,
-    __name__ + ".QtHelp": QtHelp,
-    __name__ + ".QtCompat": QtCompat,
-})
+    """
 
+    _setup("PySide2", ["QtUiTools"])
 
-"""
+    Qt.__binding_version__ = __import__("PySide2").__version__
+
+    if hasattr(Qt, "QtUiTools"):
+        Qt.QtCompat.load_ui = lambda fname: \
+            Qt._QtUiTools.QUiLoader().load(fname)
+
+    if hasattr(Qt, "QtGui") and hasattr(Qt, "QtCore"):
+        Qt.QtCore.QStringListModel = Qt._QtGui.QStringListModel
 
-Special case
+    if hasattr(Qt, "QtWidgets"):
+        Qt.QtCompat.setSectionResizeMode = \
+            Qt._QtWidgets.QHeaderView.setSectionResizeMode
 
-In some bindings, members are either misplaced or renamed.
+    if hasattr(Qt, "QtCore"):
+        Qt.__qt_version__ = Qt._QtCore.qVersion()
+        Qt.QtCompat.translate = Qt._QtCore.QCoreApplication.translate
 
-TODO: This is difficult to read, compared to the above dictionary.
-    Find a better way of implementing this, that also simplifies
-    adding or removing members.
+        Qt.QtCore.Property = Qt._QtCore.Property
+        Qt.QtCore.Signal = Qt._QtCore.Signal
+        Qt.QtCore.Slot = Qt._QtCore.Slot
+
+        Qt.QtCore.QAbstractProxyModel = Qt._QtCore.QAbstractProxyModel
+        Qt.QtCore.QSortFilterProxyModel = Qt._QtCore.QSortFilterProxyModel
+        Qt.QtCore.QItemSelection = Qt._QtCore.QItemSelection
+        Qt.QtCore.QItemSelectionModel = Qt._QtCore.QItemSelectionModel
+
+
+def _pyside():
+    """Initialise PySide"""
+
+    _setup("PySide", ["QtUiTools"])
+
+    Qt.__binding_version__ = __import__("PySide2").__version__
```

Comment:
hey - i think this should be 
Qt.__binding_version__ = __import__("PySide").__version__